shop/src/main/java/ro/msg/learning/shop/strategy/MostAbundantSelectionStrategy.java

```diff
@@ -0,0 +1,94 @@
+package ro.msg.learning.shop.strategy;
+
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import ro.msg.learning.shop.domain.Customer;
+import ro.msg.learning.shop.domain.Location;
+import ro.msg.learning.shop.domain.Order;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.dto.OrderCreateDto;
+import ro.msg.learning.shop.dto.ProductIdAndQuantityDto;
+import ro.msg.learning.shop.exception.ResourceNotFoundException;
+import ro.msg.learning.shop.repository.CustomerRepository;
+import ro.msg.learning.shop.repository.OrderRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.ProductService;
+import ro.msg.learning.shop.service.StockService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static ro.msg.learning.shop.service.OrderService.CUSTOMER_NOT_FOUND;
+import static ro.msg.learning.shop.service.StockService.PRODUCT_NOT_FOUND_WITH_ID;
+
+
+@AllArgsConstructor
+public class MostAbundantSelectionStrategy implements LocationSelectionStrategy {
+
+    public static final String INSUFFICIENT_STOCK_FOR_PRODUCT_IN_LOCATION = "Insufficient stock for product in location";
+    @Autowired
+    private StockService stockService;
+    @Autowired
+    private StockRepository stockRepository;
+    @Autowired
+    private ProductService productService;
+    @Autowired
+    private CustomerRepository customerRepository;
+    @Autowired
+    private OrderRepository orderRepository;
+
+    @Override
+    public Order selectLocations(OrderCreateDto orderCreateDto) {
+        List<Object[]> locationsMostAbundant = stockRepository.findLocationsMostAbundant(convertToProductList(orderCreateDto));
```

Comment:
size does matter.
please refactor this into tinier and easier to read methods

Reply:
Refactored.